src/Symfony/Bundle/TwigBundle/Resources/config/twig.xml

```diff
@@ -59,6 +59,9 @@
             <argument type="service" id="twig" />
             <argument type="service" id="templating.name_parser" />
             <argument type="service" id="templating.locator" />
+
+            <autowiring-type>Symfony\Component\Templating\EngineInterface</autowiring-type>
+            <autowiring-type>Symfony\Bundle\FrameworkBundle\Templating\EngineInterface</autowiring-type>
```

Comment:
if both engines are enabled, the delegating engine should be used, as this is what the ``templating`` service will be.

Btw, if you configure only the PHP engine, your PR is also broken as the delegating engine won't be used either for the ``templating`` service.

This config should be set in the compiler pass defining what the ``templating`` service is